src/Admin/ReenableCssTransientCachingAjaxAction.php

```diff
@@ -56,14 +56,15 @@ public function register() {
 	 * @param string $hook_suffix Hook suffix to identify from what admin page the call is coming from.
 	 */
 	public function register_ajax_script( $hook_suffix ) {
-		if ( 'site-health.php' !== $hook_suffix ) {
+		$supported_hook_suffix = defined( 'HEALTH_CHECK_PLUGIN_VERSION' ) ? 'tools_page_health-check' : 'site-health.php';
+		if ( $supported_hook_suffix !== $hook_suffix ) {
 			return;
 		}
 
 		$script = <<< 'JS_SCRIPT'
 ;( function () {
-	window.addEventListener( 'DOMContentLoaded', ( event ) => {
-		var selector = SELECTOR;
+	var selector = SELECTOR;
+	setTimeout( function () {
```

Comment:
Is the problem rather that the `DOMContentLoaded` event is being added to the `window` as opposed to the `document`? Or what if the `load` event on the `window` were used instead?

Reply:
The problem is that the `DOMContentLoaded` may fire before the site health test is added to the page. I guess the button was working before because the site health test was being added just in time before this event listener fires, and now that there are direct tests being added before this test, the event listener now fires before the site health test is added to the page.

Reply:
Switching this instead to run at the `load` event of `window` should do the trick, since it always runs after all `DOMContentLoaded` event handlers have run.

Reply:
The issue lies between the time the site health test DOM element is [added to the page](https://github.com/WordPress/wordpress-develop/blob/d4abd5ca4a500f7b704c59d7a03ac1c583fde6a3/src/js/_enqueues/admin/site-health.js#L362-L364) and when our JS executes. Both `DOMContentLoaded` and `load` events may have already surpassed by the time the site health is added to the page unfortunately.

Reply:
It looks like that code is running in `jQuery( function( $ ) {})` which is the same as `DOMContentLoaded`. It doesn't work when switching to use the `load` event on `window`?

Reply:
Hmm it works in Chrome but not Firefox. The button is still being inserted after the `load` event handler callback executes.

Reply:
Yeah, I was getting inconsistent results as well. I think any timeout is going to be problematic and that a different approach is needed that doesn't rely on the element being in the DOM. Namely, this should have used event delegation from the start. Please try again after c57773518d15755b3726f741420dbefc52efeafb.